cpp/examples/protobuf_pointcloud_writer/protobuf_pointcloud_writer.cpp

```diff
@@ -0,0 +1,153 @@
+#define MCAP_IMPLEMENTATION
+
+#include "descriptor.pb.h"
+#include "foxglove/PointCloud.pb.h"
+#include "mcap/writer.hpp"
+#include <chrono>
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <random>
+
+#define NS_PER_MS 1000000
+#define NS_PER_S 1000000000
+#define POINTS_PER_CLOUD 1000
+#define FIELDS_PER_POINT 3
+
+// Point represents a point in 3D space.
+struct Point {
+  float x;
+  float y;
+  float z;
+};
+
+// PointGenerator generates random points on a sphere.
+class PointGenerator {
+private:
+  std::mt19937 _generator;
+  std::uniform_real_distribution<float> _distribution;
+
+public:
+  PointGenerator(uint32_t seed = 0)
+      : _generator(seed)
+      , _distribution(0.0, 1.0) {}
+
+  // next produces a random point on the unit sphere, scaled by `scale`.
+  Point next(float scale) {
+    float theta = 2 * M_PI * _distribution(_generator);
+    float phi = acos(1 - 2 * _distribution(_generator));
+    Point point;
+    point.x = float((sin(phi) * cos(theta)) * scale);
+    point.y = float((sin(phi) * sin(theta)) * scale);
+    point.z = float(cos(phi) * scale);
+    return point;
+  }
+};
+
+// WriteLittleEndianFloat writes a little-endian float into the `output` string of bytes at
+// `offset`.
+void WriteLittleEndianFloat(float input, std::string* output, size_t offset) {
+  static_assert(sizeof(uint32_t) == sizeof(float));
+  uint32_t asInt = *reinterpret_cast<uint32_t*>(&input);
+  if (output->size() <= offset + 3) {
+    std::cerr << "tried to write 4 bytes at offset " << offset << " into buffer of size "
+              << output->size() << std::endl;
+    abort();
+  }
+  // write in little-endian order, no matter what endian-ness the host uses.
+  // the LSB goes into the first byte.
+  (*output)[offset + 0] = asInt & 0xFF;
+  (*output)[offset + 1] = (asInt >> 8) & 0xFF;
+  (*output)[offset + 2] = (asInt >> 16) & 0xFF;
+  (*output)[offset + 3] = (asInt >> 24) & 0xFF;
+}
+
+int main(int argc, char** argv) {
+  if (argc != 2) {
+    std::cerr << "Usage: " << argv[0] << " <output.mcap>\n";
+    return 1;
+  }
+  const char* outputFilename = argv[1];
+
+  mcap::McapWriter writer;
+  auto options = mcap::McapWriterOptions("x-protobuf");
+  std::ofstream out(outputFilename, std::ios::binary);
+  writer.open(out, options);
+  // set up the schema and channel.
+  mcap::Schema schema("foxglove.PointCloud", "protobuf",
+                      std::string_view((char*)(descriptor_pb_bin), descriptor_pb_bin_len));
```

Comment:
Not sure where these `descriptor_pb_bin` are supposed to come from right now, but I think we should use @danchia's method to build the descriptor set at runtime like this instead: https://foxglove.dev/blog/recording-robocar-data-with-mcap

It seems like less of a hassle than having to run with `--descriptor_set_out` and ship that data along with the binary.